File: spring project/animalshome/src/main/java/fmi/course/hcmi/animalshome/dto/PetDetailsDto.java
package fmi.course.hcmi.animalshome.dto;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@ToString
@Getter
@Setter
public class PetDetailsDto {
    @NotBlank
    @Size(min = 2, max = 50, message = "The name of the pet must be between 2 and 50 characters long!")
    private String petName;

    @NotBlank
    private String breed;

    @NotBlank
    private String color;

    @NotNull
    private String city;

    @Min(value = 0)
    @Max(value = 20)
    private int age;

    @Min(value = 0)
    @Max(value = 11)
    private int months;

    @NotNull
    private Gender gender;

    @Min(value = 1)
    @Max(value = 500)
    private double weight;

    @NotNull
    private PetType petType;
}

File: frontend/Animals-Home/src/app/model/petDetailsDto.ts
import { Gender } from './gender';
import { PetType } from './petType';

export class PetDetailsDto {
     petName: string;
     breed: string;
     color: string;
     city: string;
     age: number;
     months: number;
     gender: Gender;
     weight: number;
     petType: PetType;

     public setPetName(petName: string) {
          this.petName = petName;
     }

     public setBreed(breed: string) {
          this.breed = breed;
     }

     public setColor(color: string) {
          this.color = color;
     }

     public setCity(city: string) {
          this.city = city;
     }

     public setAge(age: number) {
          this.age = age;
     }

     public setMonths(months: number) {
          this.months = months;
     }

     public setGender(gender: Gender) {
          this.gender = gender;
     }

     public setWeight(weight: number) {
          this.weight = weight;
     }

     public setPetType(petType: PetType) {
          this.petType = petType;
     }
}